File: libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo/runtime/java/network/NetworkTransport.java
package com.apollographql.apollo.runtime.java.network;

import com.apollographql.apollo.api.ApolloRequest;
import com.apollographql.apollo.api.Operation;
import com.apollographql.apollo.runtime.java.ApolloCallback;
import com.apollographql.apollo.runtime.java.ApolloDisposable;
import org.jetbrains.annotations.NotNull;

public interface NetworkTransport {
  <D extends Operation.Data> void execute(@NotNull ApolloRequest<D> request, @NotNull ApolloCallback<D> callback, ApolloDisposable disposable);

  void dispose();
}

File: apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo/network/NetworkTransport.kt
package com.apollographql.apollo.network

import com.apollographql.apollo.api.ApolloExperimental
import com.apollographql.apollo.api.CustomScalarAdapters
import com.apollographql.apollo.api.ExecutionContext
import com.apollographql.apollo.api.Operation
import com.apollographql.apollo.ApolloRequest
import com.apollographql.apollo.interceptor.ApolloResponse
import kotlinx.coroutines.flow.Flow

@ApolloExperimental
interface NetworkTransport {

  fun <D : Operation.Data> execute(
      request: ApolloRequest<D>,
      customScalarAdapters: CustomScalarAdapters,
      executionContext: ExecutionContext
  ): Flow<ApolloResponse<D>>
}
